### DISCUSSION TITLE
How to replace content before vitepress process it?

### DISCUSSION THREAD

**Discussion by HazzazBinFaiz** [vuejs/vitepress#1324] [Q&A]
We have a collection of markdown files organized in directory.

We've used a little code snippet at the bottom of each markdown page for some metadata. like below

<pre>
```yaml
title: Title of the page
contributors: Mr. X, Mr. Y
```
</pre>

I want to convert the title to Frontmatter title and pass contributor name(s) in a component. like below

```
---
title: Title of the page
---

<Contributors names="Mr. X, Mr. Y" />
```

How can I replace those content of file before vitepress process it.

**Comment by brc-dd** [comment]
Write some markdown-it plugin for that.
